x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/related_integrations/related_integration_field.tsx

```diff
@@ -0,0 +1,205 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ChangeEvent } from 'react';
+import React, { useCallback, useMemo } from 'react';
+import { capitalize } from 'lodash';
+import type { EuiComboBoxOptionOption } from '@elastic/eui';
+import {
+  EuiTextTruncate,
+  EuiButtonIcon,
+  EuiComboBox,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiPanel,
+} from '@elastic/eui';
+import type { FieldHook } from '../../../../shared_imports';
+import type { Integration, RelatedIntegration } from '../../../../../common/api/detection_engine';
+import { useIntegrations } from '../../../../detections/components/rules/related_integrations/use_integrations';
+import { IntegrationStatusBadge } from './integration_status_badge';
+import * as i18n from './translations';
+
+interface RelatedIntegrationItemFormProps {
+  field: FieldHook<RelatedIntegration>;
+  relatedIntegrations: RelatedIntegration[];
+  onRemove: () => void;
+}
+
+export function RelatedIntegrationField({
+  field,
+  relatedIntegrations,
+  onRemove,
+}: RelatedIntegrationItemFormProps): JSX.Element {
+  const { data: integrations, isInitialLoading } = useIntegrations();
+  const [integrationOptions, selectedIntegrationOptions] = useMemo(() => {
+    const currentKey = getKey(field.value.package, field.value.integration);
+    const relatedIntegrationsButCurrent = relatedIntegrations.filter(
+      (ri) => getKey(ri.package, ri.integration) !== currentKey
+    );
+    const unusedIntegrations = filterOutUsedIntegrations(
+      integrations ?? [],
+      relatedIntegrationsButCurrent
+    );
+
+    const options = unusedIntegrations.map(transformIntegrationToOption) ?? [];
+    const fallbackSelectedOption =
+      field.value.package.length > 0
+        ? {
+            key: currentKey,
+            label: `${capitalize(field.value.package)} ${field.value.integration ?? ''}`,
+          }
+        : undefined;
+    const selectedOption =
+      options.find((option) => option.key === currentKey) ?? fallbackSelectedOption;
+
+    return [options, selectedOption ? [selectedOption] : []];
+  }, [integrations, field.value, relatedIntegrations]);
+
+  const [packageErrorMessage, versionErrorMessage] = useMemo(() => {
+    const packagePath = `${field.path}.package`;
+    const versionPath = `${field.path}.version`;
+
+    return [
+      field.errors.find((err) => 'path' in err && err.path === packagePath),
+      field.errors.find((err) => 'path' in err && err.path === versionPath),
+    ];
+  }, [field.path, field.errors]);
+
+  const handleIntegrationChange = useCallback(
+    ([changedSelectedOption]: Array<EuiComboBoxOptionOption<Integration>>) =>
+      field.setValue({
+        package: changedSelectedOption?.value?.package_name ?? '',
+        integration: changedSelectedOption?.value?.integration_name,
+        version: changedSelectedOption?.value
```

Comment:
You could probably simplify/clarify this logic with a helper function: 

```ts
const addCaretToVersion = (version: string | undefined): string => (version ? `^${version}` : '');
```

Reply:
I ended up adding moving part of the logic to `calculateRelevantSemver()`. It definitely improves readability.